### ISSUE TITLE
EDDI starts up but fails to show GUI

### ISSUE THREAD

**Issue by richardbuckle** [EDCD/EDDI#1159]
## EDDI version in which issue found
3.3.4-rc3 standalone

## Steps to reproduce
1. See reports from Tazpoken in EDCD Discord, EDDI channel.
2. Apparently due to a trial TTS voice licence period expiring.

## Expected
EDDI starts up and shows GUI.

## Observed
EDDI starts up and is visible in Task Manager but GUI never appears.

## Investigation
Logs attached.

**Comment by richardbuckle** [comment]
[Tazpoken_EDDI_logs.zip](https://github.com/EDCD/EDDI/files/2834238/Tazpoken_EDDI_logs.zip)

**Comment by Tkael** [comment]
Unfortunately, nothing in these logs seems to suggest a problem with an expired TTS voice. 
I do see some connectivity related errors, particularly with respect to the EDSM configuration.
It's possible that the expired TTS voice was unrelated to the actual failure to launch?

**Comment by DianonForce** [comment]
Hi, i have the same problem, when i delete all in %appdata% eddi the Gui comes up again for one time. After closing and restart, the gui dosn't come back. I can reproduce this.
All work fine until yesterday, today i experimented with some alternative TTS voices an install this trail voice from https://harposoftware.com/en/german/216-Anna-Nuance-Voice.html.
So i think richards idea that it have something to do with a trail TTS voice could be right 

I'm useing EDDI 3.3.4 rc3

**Comment by Tkael** [comment]
I'm glad it's working. :-) Would you please install [the latest version released today](https://github.com/EDCD/EDDI/releases/download/Release%2F3.3.4-rc4/EDDI-3.3.4-rc4.exe) and let us know if the problem recurs? 
If the issue recurs, respond here and we'll re-open the issue.

**Comment by hoksilato2** [comment]
This is closed but, I have the same problem after reset windows 10 configuration. After install 3.3.4 version (the last) GUI never appears but task is active. I only install VoiceAttack and EDDI.

**Comment by hoksilato2** [comment]
[eddi.log](https://github.com/EDCD/EDDI/files/2848117/eddi.log)

Log has alot of warnings :(

**Comment by hoksilato2** [comment]
Ok, this is a serious bug. If I delete %appdata%\EDDI folder, UI appears. But, only first time. here the last log:
[eddi.log](https://github.com/EDCD/EDDI/files/2848121/eddi.log)

**Comment by hoksilato2** [comment]
Ok, more testing: The problem is missionmonitor.json. If I delete this file GUI works again, but I need delete this file every EDDI's start :(

**Comment by Tkael** [comment]
Do you have pending / incomplete missions?

**Comment by hoksilato2** [comment]
Yes, with negative count. I'm out of the game some days...

**Comment by Tkael** [comment]
Negative count? What do you mean?

**Comment by hoksilato2** [comment]
I have one mission expired with -6D count.

**Comment by Tkael** [comment]
If you are able to enter the game and clear the mission, does the problem go away?

**Comment by hoksilato2** [comment]
omg yes! Well it's a little strange because the missions in the game was not the same that the monitor mission but, In game, I delete all missions and then deleted json too. Now EDDI GUI works again.

Tried three times. EDDI's mission monitor works ok now.

**Comment by Tkael** [comment]
Ok. If you could fetch your missionmonitor.json file from the recycle bin and pm it to @Hoodathunk, I believe he's been collecting data on this issue.

**Comment by Hoodathunk** [comment]
Ideally, I need to see the missionmonitor.json file, associated journal events from the log, and the any pertinent eddi.log data.

**Comment by silverbane1970** [comment]
i cannot find the missionmonitor.json file

**Comment by Tkael** [comment]
Ok. Then please attach `eddi.log`and your latest journal file.

**Comment by silverbane1970** [comment]
i found the missionmonitor.log file it is working now

**Comment by Tkael** [comment]
We think we've identified and found a fix for this issue. The temporary resolution is to remove missionmonitor.json from the `%APPDATA%\EDDI` folder.

**Comment by hoksilato2** [comment]
[missionmonitor.json.log](https://github.com/EDCD/EDDI/files/2848635/missionmonitor.json.log)
This is the missionmonitor.json I needed delete.

My eddi.log is in my first post. And Journal I think is the attached too. I remember you that missionjournal hadn't the same missions when I open the game. My last EDDI version was 3.3.4-RC3, and the tests I use last 3.3.4
[Journal.190201194312.01.log](https://github.com/EDCD/EDDI/files/2848637/Journal.190201194312.01.log)

**Comment by Roming22** [comment]
I had the same issue. Here's the file that caused the problem. The expired mission was the hostage one.
[missionmonitor.log](https://github.com/EDCD/EDDI/files/2849492/missionmonitor.log)

**Comment by hoksilato2** [comment]
In spanish forum we have 4 users with the same problem. Deleting missionmonitor.json problem fixed but in some cases if I close EDDI in the middle of the game, problem appears again.

**Comment by JorgenSmith** [comment]
Ah, here's my failing missionmonitor.json file - this was a failed mission. It works now having removed it. Version 3.3.4. Cheers!

[missionmonitor.json.log](https://github.com/EDCD/EDDI/files/2851247/missionmonitor.json.log)

**Comment by trasa** [comment]
I'm having this same problem, starting today, with 3.3.4. EDDI started and told me I had 16 pending missions, which was incorrect. And then EDDI ended up in the gui-less hung state as others have reported.

Deleted %appdata%/roaming/eddi/missionmonitor.json and the app starts up correctly again.

[missionmonitor.json.bad.log](https://github.com/EDCD/EDDI/files/2853651/missionmonitor.json.bad.log)

**Comment by Hoodathunk** [comment]
@trasa, I have a pretty good idea what's going on.

1. We added functionality to re-parse your last game session from the journal log to improve accuracy of reporting scans to EDDN/EDSM. Which worked... EDDN error reporting is at an all time low.
2. We also improved event triggering to ensure system scans were processed in the proper order and to prevent repeat scan reports, which has also shown good results.
3. Re-parsing the journal log had an unintended consequence of causing duplicate mission entries in the mission log. In 3.3.4, we added protection to prevent duplicates and we time stamped the missionmonitor.json file to prevent re-parsing (to speed up the Log-Load process) and add missed mission-related events. Running 3.3.4, the first time would start with a min timestamp and add previous `Mission accepted' events, but immediately removed them due to subsequent `Mission completed` events. This would only occur the 1st time, as the json time stamp would now be up to date.

Now for the not so good part...

4. Due to the switch to a queue-based processing of journal events (to improve scanning), this caused some threads to become vulnerable to a 'deadlock' condition, when competing for the same resource. This deadlock, in particular, presented itself when processing the derived `Mission expired` event, essentially locking EDDI up... hard.

So what happened, in your case (and tangentially others), the 'Log-Load' parsing added the (previously completed) events, and then immediately identified them as 'expired', which resulted in EDDI 'freezing'.

Good news we have corrected the deadlock issue and, pending some additional testing, we expect to have an update tomorrow.
